### PR TITLE
Fit overlay within viewport boundary

### PR DESCRIPTION
**Pull request by taion**
For react-bootstrap/react-bootstrap#173

I have no idea how to write automated tests for this.

ETA: Tests added, thanks @AlexKVal for suggesting how to write them.

### CODE REVIEW

**Comment by AlexKVal** [comment]
Just idea about tests.
It should be sufficient at least to test `calcOverlayPosition()` return values for several cases.
Better than nothing and it will cover new code.

Disclaimer: I did not analyze it thoroughly, so I may propose something unrealizable.
:cherries:

**Comment by taion** [comment]
The positioning depends on viewport size for the most common case, though. How does that work with Karma?

**Comment by AlexKVal** [comment]
In previous post I meant to mock browser functionality and just check that function calculates the right numbers.
But as it turned out it is possible to change `viewportSize` and browser `window` size.

I hope it would help:
from questions about `how to test responsive design`
`karma.conf.js`
```js
customLaunchers: {
  'Chrome_Desktop': {
    base: 'Chrome',
      options: {
        viewportSize: {
          width: 1228,
          height: 1000
      }
    }
  }
},
```
```js
module.exports = function(config) {
  config.set({
    browsers: ['mobile', 'desktop'],
    customLaunchers: {
      mobile: {
        base: "Chrome",
        flags: "--window-size=320,600"
      },
      desktop: {
        base: "Chrome",
        flags: "--window-size=700,600"
      }
    }
  });
}
```

**Comment by taion** [comment]
Oh, I see what you mean. I'll use the mocks for now.

File: src/OverlayTrigger.js [hunk 1]
```diff
@@ -33,13 +33,15 @@ const OverlayTrigger = React.createClass({
     delayShow: React.PropTypes.number,
     delayHide: React.PropTypes.number,
     defaultOverlayShown: React.PropTypes.bool,
-    overlay: React.PropTypes.node.isRequired
+    overlay: React.PropTypes.node.isRequired,
+    containerPadding: React.PropTypes.number
```
Comment (mtscout6) [inline src/OverlayTrigger.js]: With this added prop can you include some documentation for it so other's know how/when to use it.

**Comment by mtscout6** [comment]
When you wrap this up if you could amend the commit message with `[fixed]` so this bug fix is reflected in the changelog.

**Comment by taion** [comment]
I'll add it to the existing (broken) "scrolling popover" example. Will need a bit of time to get the test cases set up, though.

**Comment by taion** [comment]
Added explicit `containerPadding` example, as well as tests for `calcOverlayPosition`.

**Comment by mtscout6** [comment]
LGTM

**Comment by AlexKVal** [comment]
Test are :+1:  You really did it :cherries:

**Pull request merged**